src/phpDocumentor/Reflection/Php/Factory/File.php

```diff
@@ -61,6 +65,30 @@ public function __construct(NodesFactory $nodesFactory, Adapter $adapter = null)
 
         $this->nodesFactory = $nodesFactory;
         $this->adapter = $adapter;
+
+        $this->middlewareChain = $this->createExecutionChain($middleware);
+    }
+
+    /**
+     * @param Middleware[] $middlewareList
+     *
+     * @return callable
+     */
+    private function createExecutionChain($middlewareList)
+    {
+        $lastCallable = function($command) {
+            return $this->createFile($command);
+        };
+
+        while ($middleware = array_pop($middlewareList)) {
+            if (! $middleware instanceof Middleware) {
+                throw new InvalidArgumentException();
```

Comment:
Can you also provide a message with this exception that mentions why this failed?